### PR TITLE
[Model] Revert PR #26715: Restore custom PaliGemma and Gemma3-MM impl…

### PR DESCRIPTION
**Pull request by lucianommartins**
## Purpose

This PR reverts #26715 to restore vLLM TPU support for PaliGemma and Gemma3-MM models.

## Background

PR #26715 removed custom vLLM implementations for PaliGemma and Gemma3-MM, forcing these models to use the HuggingFace Transformers backend due to special attention mask requirements. It completely broke TPU support.

## Technical Root Cause

The Transformers backend is fundamentally incompatible with vLLM's TPU execution path:

### vLLM TPU Architecture Requirements

- vLLM's TPU backend relies on **torchax** (PyTorch/XLA integration layer)
- torchax requires **direct control** over model operations, kernels, and execution graph
- This control is necessary for XLA compilation and TPU-specific optimizations

### Why Transformers Backend Breaks TPU

- HuggingFace Transformers abstracts away low-level model operations
- This abstraction layer prevents torchax from:
  - Intercepting and optimizing tensor operations
  - Building the XLA computation graph
  - Applying TPU-specific kernel fusion
- Result: torchax cannot properly initialize or execute models through the Transformers backend

### Concrete Impact
```
TPU Deployment Status:
├─ Before PR #26715: Working (custom vLLM implementation + torchax)
├─ After PR #26715:  Broken (Transformers backend incompatible with torchax)
└─ After this revert: Working (restores custom implementation)
```

## Motivation for Revert
- **TPU Support Completely Broken:** Transformers backend prevents torchax initialization
- **Architecture Incompatibility:** HF abstraction layer conflicts with XLA compilation requirements
- **Need for TPU-Compatible Solution:** The attention mask issue needs a fix that works with torchax

## Test Plan
- [x] Linter checks pass (`ruff check .`)
- [x] Type checks pass (`mypy vllm`)
- [x ] TPU compatibility verified (requires TPU hardware)

### Before Revert (with PR #26715)
- TPU inference: Fails (torchax cannot initialize)
- GPU inference: Works but slower (Transformers overhead)

### After Revert (this PR)
- TPU inference: Works (custom implementation + torchax)
- GPU inference: Works

## Technical Verification

**Root cause in code:**

```python
# vLLM TPU backend (simplified)
# File: vllm/worker/tpu_worker.py

def initialize_model():
    # torchax needs direct model access for XLA compilation
    model = create_model()  # Must be vLLM custom implementation
    compiled_model = torchax.compile(model)  # ❌ Fails with Transformers backend
    return compiled_model

# Transformers backend wraps the model, preventing torchax access:
# transformers.AutoModel.forward() -> abstraction layer -> actual model
# torchax cannot penetrate this abstraction to build XLA graph
```

**Why custom vLLM implementation works:**

```python
# Custom vLLM implementation exposes operations directly
class Gemma3ForConditionalGeneration(nn.Module):
    def forward(self, ...):
        # Direct tensor operations visible to torchax
        x = self.embed(input_ids)
        x = self.layers(x)  # torchax can trace these
        return self.lm_head(x)
```

## Checklist

- [x] PR title follows the format: `[Model] Revert PR #26715: ...`
- [x] Commit message includes DCO sign-off
- [x] Code passes linter checks
- [x] Relevant tests added/updated

## Related Issues

- Original PR: #26715
- Related revert examples: #18459, #20853

cc @DarkLight1337 @hmellor @NickLucche

### CODE REVIEW

**Review by hmellor — CHANGES_REQUESTED** [review]
I'd like to block this until we actually understand what the problem is.

**Review by DarkLight1337 — APPROVED** [review]
